### DISCUSSION TITLE
Packaged app doesn't request automation access (macOS)

### DISCUSSION THREAD

**Discussion by need4swede** [pyinstaller/pyinstaller#6932] [Help] [answered]
My program uses the appscript library to automate mailing tasks. 

When running the app through the executable, 'Terminal' requests permission to control Mail.app. Once granted, the program functions. Likewise, if I run my code through VSCode, it is VSCode that requests permission to control Mail.app. 

However, if I run my script through the pyinstaller packaged app, it doesn't ask for permission and the app simply crashes.

## Sample

```
from appscript import app, k

mail = app('Mail')

msg = mail.make(
    new=k.outgoing_message,
    with_properties={
        k.subject: 'RE: pyinstaller'})

msg.activate()
```

## Error message if the app isn't granted permission:

> OSERROR: -1743
> MESSAGE: The user has declined permission.
> COMMAND: app('/System/Applications/Mail.app'

How do I excplitly request permission from macOS through my packaged app?

Thank you!

**Comment by bwoodsend** [comment]
You can't. If macOS's SIP doesn't behave itself out of the box, the best you can do is catch the exception and turn it into a prompt telling the user to go to Sytem Preferences -> Security -> ...

**Reply by need4swede** [reply]
rokm was able to solve my inquiry - but I appreciate your time and input! Thank you, kind stranger

**Comment by rokm** [comment, marked as answer]
Take a look at #6517; in particular these two comments:
- https://github.com/orgs/pyinstaller/discussions/6517#discussioncomment-1992866
- https://github.com/orgs/pyinstaller/discussions/6517#discussioncomment-2027674

**Reply by need4swede** [reply]
rokm,

You are an absolute genius, my friend. This definitely resolved my problem.
Thank you so very much!
